MathlibTest/DeprecatedSyntaxLinter.lean

```diff
@@ -77,3 +82,69 @@ warning: declaration uses 'sorry'
 -/
 #guard_msgs in
 example : False := by admit
+
+set_option linter.style.maxHeartbeats true
+/--
+warning: Please, add a comment explaining the need for modifying the maxHeartbeat limit, as in
+set_option maxHeartbeats 10 in
+-- reason for change
+...
+ note: this linter can be disabled with `set_option linter.style.maxHeartbeats false`
+-/
+#guard_msgs in
+set_option maxHeartbeats 10 in
+section
+
+/--
+warning: Please, add a comment explaining the need for modifying the maxHeartbeat limit, as in
+set_option maxHeartbeats 10 in
+-- reason for change
+...
+ note: this linter can be disabled with `set_option linter.style.maxHeartbeats false`
+-/
+#guard_msgs in
+set_option maxHeartbeats 10 in
+set_option maxHeartbeats 10 in
+section
+
+/--
+warning: Please, add a comment explaining the need for modifying the maxHeartbeat limit, as in
+set_option synthInstance.maxHeartbeats 10 in
+-- reason for change
+...
+ note: this linter can be disabled with `set_option linter.style.maxHeartbeats false`
+-/
+#guard_msgs in
+set_option synthInstance.maxHeartbeats 10 in
+section
+
+/--
+warning: Please, add a comment explaining the need for modifying the maxHeartbeat limit, as in
+set_option maxHeartbeats 10 in
+-- reason for change
+...
+ note: this linter can be disabled with `set_option linter.style.maxHeartbeats false`
+-/
+#guard_msgs in
+set_option maxHeartbeats 10 in
+set_option synthInstance.maxHeartbeats 10 in
+/- The comment here is not enough to silence the linter:
+the *first* `maxHeartbeats` option should have a comment. -/
+section
```

Comment:
For paraoina or readers who just read this test, not the implementation: add another version where the outer set_option has a comment, but the inner one does not?

Reply:
Comment added!